langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingModel.java

```diff
@@ -26,12 +37,40 @@ protected String getModelId() {
         return model;
     }
 
+    /**
+     * 1024 is default size of output vector for Titan Embedding model V2.
+     */
+    @Builder.Default
+    private final Integer dimensions = 1024;
+
+    /**
+     *  A flag indicating whether or not to normalize the output embeddings.
+     *  It defaults to true, which is optimal for RAG use cases.
+     */
+    @Builder.Default
+    private final Boolean normalize = true;
+
+
     @Override
     protected List<Map<String, Object>> getRequestParameters(List<TextSegment> textSegments) {
-        return textSegments.stream()
-                .map(TextSegment::text)
-                .map(text -> of("inputText", text))
-                .collect(Collectors.toList());
+        if (MODEL_V1_ID.equals(this.model)) {
+            return textSegments.stream()
+                    .map(TextSegment::text)
+                    .map(text -> of("inputText", text))
+                    .collect(Collectors.toList());
+        } else if (MODEL_V2_ID.equals(this.model)) {
+            List<Map<String, Object>> result = new ArrayList<>();
+            for (TextSegment textSegment : textSegments) {
+                Map<String, Object> parameters = new HashMap<>();
+                parameters.put("inputText", textSegment.text());
+                parameters.put("dimensions", dimensions);
+                parameters.put("normalize", normalize);
+                result.add(parameters);
+            }
+            return result;
+        } else {
+            throw new IllegalArgumentException("Unsupported model: " + this.model);
```

Comment:
When the new model will come out, users will have to wait untill we add it explicitly and release the next version of LC4j to be able to use it...

Reply:
Already addressed with one previous comment. Totally agree, we should think of V3, V4,etc.